src/libertem/io/dataset/raw.py

```diff
@@ -201,6 +202,9 @@ def __init__(self, path, scan_size, dtype, detector_size=None, enable_direct=Fal
                                  "please use EMPAD DataSet")
             detector_size = crop_detector_to
 
+        if detector_size is None:
+            raise ValueError("missing 1 required argument: 'detector_size'")
```

Comment:
This should be a `TypeError`; see the descriptions on https://docs.python.org/3/library/exceptions.html, especially:

```
exception ValueError
Raised when an operation or function receives an argument that has the right type but an inappropriate value, and the situation is not described by a more precise exception such as IndexError
```

Also, as you noted in #600, `TypeError` is also the exception type you get in all other cases if you omit a required parameter.